### ISSUE TITLE
BUG: Unhandled error in plugin discover: Expecting value: line 1 column 1 (char 0)

### ISSUE THREAD

**Issue by programatix** [Flexget/Flexget#1973]
### Expected behaviour:
Discover should run normally.

### Actual behaviour:
Discover crashed.

### Steps to reproduce:
- Step 1: Run task "download-movies-discover"

#### Config:
```
  movies-metainfo:
    metainfo_movie: yes
    tmdb_lookup: yes
    parsing:
      movie: guessit

  reject:
    regexp:
      reject:
        - \b(s|d)ub(s|bed|lado|titulado)?\b
        - \bhc\b
        - \bdual\b
        - \b3d\b
        - \bfre(nch)?\b
        - \bita(lian)?\b
        - \bspa(nish)?\b
        - \bger(man)?\b
        - \bcastellano\b
        - \brus(sian)?\b
        - \bswe(dish)?\b
        - \bhindi\b
        - '[\. ]korsub[\. ]?'

  movie:
    torrent_alive: 5                 # Will reject results with less than 1 seed
    quality: 1080p webdl+ aac+   # Make sure no screeners or cams are downloaded
    content_size:
      min: 600
      max: 7168
      strict: no
    set:
      path: '/{? folder.root ?}{? folder.moviesDVD ?}'
    template: transmissionRPC
    transmission:
      addpaused: no
      ratio: 0

  clean-movies-queue:
    priority: 31
    template:
      - disable-seen-retry
      - movies-metainfo
    filesystem:
      path:
        - "/{? folder.root ?}{? folder.moviesDVD ?}"
      recursive: yes
      retrieve: files
      regexp: '.*\.(avi|mkv|mp4)$'
    accept_all: yes
    list_remove:
      - movie_list: movies
      - trakt_list:
          username: "{? trakt.username ?}"
          account: "{? trakt.account ?}"
          list: "{? trakt.list ?}"
          type: movies

  download-movies-discover:
    priority: 33 # run after the movie list fill task
    template:
      - movies-metainfo
      - reject
      - movie
      - notifyViaEmail
    discover:
      what:
        - movie_list: movies
      from:
        - rarbg:
            category:
              - x264 720p
              - x264 1080p
            ranked: false
      interval: 1 hour
      release_estimations:
        optimistic: 75 days
    list_add:
      - subtitle_list:
          list: subtitles
          languages:
            - en
          path: '{{path}}'
          allow_dir: true
          remove_after: "4 weeks"
          check_subtitles: yes
    list_match:
      from:
        - movie_list: movies
    list_remove:
      - trakt_list:
          username: "{? trakt.username ?}"
          account: "{? trakt.account ?}"
          list: "{? trakt.list ?}"
          type: movies
```
  
#### Log:
```
2017-09-22 00:17 WARNING  movie_list    clean-movies-queue Could not get a movie name, skipping
2017-09-22 10:01 CRITICAL task          download-movies-discover BUG: Unhandled error in plugin discover: Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "/opt/lib/python3.6/site-packages/flexget/task.py", line 486, in __run_plugin
    return method(*args, **kwargs)
  File "/opt/lib/python3.6/site-packages/flexget/event.py", line 23, in __call__
    return self.func(*args, **kwargs)
  File "/opt/lib/python3.6/site-packages/flexget/plugins/input/discover.py", line 289, in on_task_input
    entries = self.estimated(entries, estimation_mode)
  File "/opt/lib/python3.6/site-packages/flexget/plugins/input/discover.py", line 201, in estimated
    est_date = estimator.estimate(entry)
  File "/opt/lib/python3.6/site-packages/flexget/plugins/estimators/est_released.py", line 28, in estimate
    estimate = estimator(entry)
  File "/opt/lib/python3.6/site-packages/flexget/plugins/estimators/est_released_movies_bluray.py", line 36, in estimate
    movie = lookup(title=movie_name, year=movie_year, session=session)
  File "/opt/lib/python3.6/site-packages/flexget/plugins/internal/api_bluray.py", line 209, in lookup
    updated_movie = BlurayMovie(title=title, year=year)
  File "<string>", line 4, in __init__
  File "/opt/lib/python3.6/site-packages/sqlalchemy/orm/state.py", line 414, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/opt/lib/python3.6/site-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/opt/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 187, in reraise
    raise value
  File "/opt/lib/python3.6/site-packages/sqlalchemy/orm/state.py", line 411, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/opt/lib/python3.6/site-packages/flexget/plugins/internal/api_bluray.py", line 77, in __init__
    search_results = bluray_request('quicksearch/search.php', **params)['items']
  File "/opt/lib/python3.6/site-packages/flexget/plugins/internal/api_bluray.py", line 34, in bluray_request
    return requests.get(full_url, params=params).json(strict=False)
  File "/opt/lib/python3.6/site-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/__init__.py", line 367, in loads
  File "/decoder.py", line 339, in decode
  File "/decoder.py", line 357, in raw_decode
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

### Additional information:

- FlexGet version: 2.10.94
- Python version: 3.6
- Installation method: pip
- Using daemon (yes/no): no

**Comment by cvium** [comment]
Any idea what the name of the movie is?

**Comment by programatix** [comment]
I have no idea as it happened intermittently. I tried running the task several times and it doesn't crash. These are the movies which are currently in my list.
```
tqqqqqnqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqnqqqqqqqqqqqqnqqqqqqqnqqqqqqqqqqqnqqqqqqqqqqqqqqqqnqqqqqqqqqqqnqqqqqqqqqu
x 501 x Aquaman                                                            x 2017       x       x           x 193968         x tt1477834 x 297802  x
x 500 x Bad Boys for Life                                                  x 2018       x       x           x 25039          x tt1502397 x 38700   x
x 499 x How to Train Your Dragon 3                                         x 2019       x       x           x 106668         x tt2386490 x 166428  x
x 498 x The Foreigner                                                      x 2017       x       x           x 225520         x tt1615160 x 379149  x
x 493 x Wind River                                                         x 2017       x       x           x 238651         x tt5362988 x 395834  x
x 492 x The Emoji Movie                                                    x 2017       x       x           x 239730         x tt4877122 x 378236  x
x 491 x The Nut Job 2: Nutty by Nature                                     x 2017       x       x           x 216553         x tt3486626 x 335777  x
x 489 x Annabelle: Creation                                                x 2017       x       x           x 239309         x tt5140878 x 396422  x
x 485 x The Hitman's Bodyguard                                             x 2017       x       x           x 232291         x tt1959563 x 390043  x
x 484 x The Batman                                                         x 2019       x       x           x 259852         x tt1877830 x 414906  x
x 483 x Gambit                                                             x 2019       x       x           x 196665         x tt3729594 x 316027  x
x 482 x Valerian and the City of a Thousand Planets                        x 2017       x       x           x 220423         x tt2239822 x 339964  x
x 481 x The Dark Tower                                                     x 2017       x       x           x 222255         x tt1648190 x 353491  x
x 480 x Han Solo: A Star Wars Story                                        x 2018       x       x           x 221890         x tt3778644 x 348350  x
x 479 x Fantastic Beasts and Where to Find Them 2                          x 2018       x       x           x 219417         x tt4123430 x 338952  x
x 478 x Jurassic World: Fallen Kingdom                                     x 2018       x       x           x 222216         x tt4881806 x 351286  x
x 475 x The Incredibles 2                                                  x 2018       x       x           x 159626         x tt3606756 x 260513  x
x 469 x Cars 3                                                             x 2017       x       x           x 159627         x tt3606752 x 260514  x
x 467 x Kung Fury                                                          x 2015       x       x           x 152524         x tt3472226 x 251516  x
x 464 x Amityville: The Awakening                                          x 2017       x       x           x 126879         x tt1935897 x 203835  x
x 463 x World War Z 2                                                      x 2019       x       x           x 216729         x tt3234552 x 336002  x
x 462 x Jumanji: Welcome to the Jungle                                     x 2017       x       x           x 222254         x tt2283362 x 353486  x
x 461 x Atomic Blonde                                                      x 2017       x       x           x 224157         x tt2406566 x 341013  x
x 460 x It                                                                 x 2017       x       x           x 242226         x tt1396484 x 346364  x
x 456 x Blade Runner 2049                                                  x 2017       x       x           x 216716         x tt1856101 x 335984  x
x 455 x Dunkirk                                                            x 2017       x       x           x 224641         x tt5013056 x 374720  x
x 445 x M:I 6 - Mission Impossible                                         x 2018       x       x           x 222160         x tt4912910 x 353081  x
x 434 x Toy Story 4                                                        x 2019       x       x           x 193972         x tt1979376 x 301528  x
x 426 x Kingsman: The Golden Circle                                        x 2017       x       x           x 223444         x tt4649466 x 343668  x
x 424 x Deadpool 2                                                         x 2018       x       x           x 226108         x tt5463162 x 383498  x
x 423 x War for the Planet of the Apes                                     x 2017       x       x           x 178749         x tt3450958 x 281338  x
x 421 x Pacific Rim: Uprising                                              x 2018       x       x           x 167163         x tt2557478 x 268896  x
x 412 x ARQ                                                                x 2016       x       x           x 258909         x tt5640450 x 410199  x
x 404 x Ant-Man and the Wasp                                               x 2018       x       x           x 223262         x tt5095030 x 363088  x
x 403 x Cyborg                                                             x            x       x           x 193918         x tt4123436 x 298622  x
x 402 x Green Lantern Corps                                                x            x       x           x 193969         x tt4120636 x 297797  x
x 401 x Shazam!                                                            x 2019       x       x           x 184992         x tt0448115 x 287947  x
x 400 x Sherlock Holmes 3                                                  x 2019       x       x           x 191791         x tt2094116 x 297708  x
x 399 x Spider-Man: Homecoming                                             x 2017       x       x           x 196213         x tt2250912 x 315635  x
x 398 x Black Panther                                                      x 2018       x       x           x 181313         x tt1825683 x 284054  x
x 394 x Walt Disney Classic Short Films Volume 1: Mickey and the Beanstalk x 2009       x       x           x 214513         x None      x 333585  x
x 393 x The Prince and the Pauper                                          x 1990       x       x           x 15648          x tt0100409 x 25831   x
x 374 x Avengers: Infinity War Part 2                                      x 2019       x       x           x 191798         x tt4154796 x 299534  x
x 373 x Avengers: Infinity War                                             x 2018       x       x           x 191797         x tt4154756 x 299536  x
x 363 x Avatar 3                                                           x 2021       x       x           x 62544          x tt1757678 x 83533   x
x 362 x Justice League 2                                                   x            x       x           x 193970         x tt4121026 x 298619  x
x 361 x Ascendant                                                          x            x       x           x 180775         x tt3663184 x 283489  x
x 360 x Flashpoint                                                         x 2020       x       x           x 195144         x tt0439572 x 298618  x
x 359 x Maze Runner: The Death Cure                                        x 2018       x       x           x 217461         x tt4500922 x 336843  x
x 357 x Captain Marvel                                                     x 2019       x       x           x 193963         x tt4154664 x 299537  x
x 354 x Avatar 2                                                           x 2020       x       x           x 56580          x tt1630029 x 76600   x
x 353 x Star Wars: Episode IX                                              x 2019       x       x           x 114335         x tt2527338 x 181812  x
x 350 x Star Wars: The Last Jedi                                           x 2017       x       x           x 114333         x tt2527336 x 181808  x
x 348 x Thor: Ragnarok                                                     x 2017       x       x           x 181312         x tt3501632 x 284053  x
x 347 x Justice League                                                     x 2017       x       x           x 94232          x tt0974015 x 141052  x
x 346 x Ratchet & Clank                                                    x 2016       x       x           x 142677         x tt2865120 x 234004  x
x 339 x The Projects: Rise of Redemption                                   x 2018       x       x           x                x tt3986678 x         x
mqqqqqvqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqvqqqqqqqqqqqqvqqqqqqqvqqqqqqqqqqqvqqqqqqqqqqqqqqqqvqqqqqqqqqqqvqqqqqqqqqj
```

**Comment by marcoacarvalho** [comment]
I'm getting the same error with this movies on my list:
Valerian and the City of a Thousand Planets (2017)
Atomic Blonde (2017)
Blade Runner 2049 (2017)
Red Sparrow (2018)
Thor: Ragnarok (2017)
Black Panther (2018)

**Comment by marcoacarvalho** [comment]
As a test I left only one movie on the list and I get the same error no matter wich movie:

```
┌1 Movies in movie list: `movie-queue`───────┬───────┬────────────────┬───────────┬─────────┐
│ # │ Movie Name    │ Movie year │ bluray_id │ rt_id │ trakt_movie_id │ imdb_id   │ tmdb_id │
├───┼───────────────┼────────────┼───────────┼───────┼────────────────┼───────────┼─────────┤
│ 1 │ Atomic Blonde │ 2017       │           │       │ 224157         │ tt2406566 │ 341013  │
└───┴───────────────┴────────────┴───────────┴───────┴────────────────┴───────────┴─────────┘
[marco@nas ~]$ flexget-dev/bin/flexget --debug execute --task movies-discover --discover-now
2017-10-21 21:54 DEBUG    manager                       Figuring out config load paths
2017-10-21 21:54 DEBUG    manager                       Adding virtualenv path
2017-10-21 21:54 DEBUG    manager                       Found config: /home/marco/.flexget/config.yml
2017-10-21 21:54 DEBUG    manager                       Config file /home/marco/.flexget/config.yml selected
2017-10-21 21:54 DEBUG    manager                       sys.defaultencoding: utf-8
2017-10-21 21:54 DEBUG    manager                       sys.getfilesystemencoding: utf-8
2017-10-21 21:54 DEBUG    manager                       flexget detected io encoding: utf-8
2017-10-21 21:54 DEBUG    manager                       os.path.supports_unicode_filenames: False
There is a FlexGet process already running for this config, sending execution there.
2017-10-21 21:54 DEBUG    manager                       Sending command to running FlexGet process: ['--debug', 'execute', '--task', 'movies-discover', '--discover-now']
2017-10-21 21:54 DEBUG    cron_env                      Encoding utf-8 stored
2017-10-21 21:54 DEBUG    util.simple_persistence                 setting key terminal_encoding value 'utf-8'
2017-10-21 21:54 DEBUG    task          movies-discover executing movies-discover
2017-10-21 21:54 DEBUG    template      movies-discover Merging template trakt-lookup into task movies-discover
2017-10-21 21:54 DEBUG    template      movies-discover Merging template content-rejects into task movies-discover
2017-10-21 21:54 DEBUG    template      movies-discover Merging template global into task movies-discover
2017-10-21 21:54 VERBOSE  discover      movies-discover Discovering 1 titles ...
2017-10-21 21:54 INFO     discover      movies-discover Ignoring interval because of --discover-now
2017-10-21 21:54 DEBUG    est_released  movies-discover Atomic Blonde (2017)
2017-10-21 21:54 DEBUG    est_movies_bluray movies-discover Searching Blu-ray.com for release date of Atomic Blonde (2017)
2017-10-21 21:54 DEBUG    api_bluray    movies-discover Cache has expired for Atomic Blonde, attempting to refresh from blu-ray.com.
2017-10-21 21:54 DEBUG    utils.requests movies-discover Fetching URL http://m.blu-ray.com/quicksearch/search.php with args () and kwargs {'params': {'section': 'bluraymovies', 'country': 'ALL', 'keyword': 'Atomic Blonde', '_': '1508633642956'}, 'allow_redirects': True, 'timeout': 30}
2017-10-21 21:54 CRITICAL task          movies-discover BUG: Unhandled error in plugin discover: Expecting value: line 1 column 1 (char 0)
2017-10-21 21:54 CRITICAL manager       movies-discover An unexpected crash has occurred. Writing crash report to /home/marco/.flexget/crash_report.2017.10.21.215403602500.log. Please verify you are running the latest version of flexget by using "flexget -V" from CLI or by using version_checker plugin at http://flexget.com/wiki/Plugins/version_checker. You are currently using version 2.10.102
2017-10-21 21:54 DEBUG    manager       movies-discover Traceback:
Traceback (most recent call last):
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/task.py", line 486, in __run_plugin
    return method(*args, **kwargs)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/event.py", line 23, in __call__
    return self.func(*args, **kwargs)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/plugins/input/discover.py", line 289, in on_task_input
    entries = self.estimated(entries, estimation_mode)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/plugins/input/discover.py", line 201, in estimated
    est_date = estimator.estimate(entry)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/plugins/estimators/est_released.py", line 28, in estimate
    estimate = estimator(entry)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/plugins/estimators/est_released_movies_bluray.py", line 36, in estimate
    movie = lookup(title=movie_name, year=movie_year, session=session)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/plugins/internal/api_bluray.py", line 209, in lookup
    updated_movie = BlurayMovie(title=title, year=year)
  File "<string>", line 4, in __init__
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/sqlalchemy/orm/state.py", line 414, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 187, in reraise
    raise value
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/sqlalchemy/orm/state.py", line 411, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/plugins/internal/api_bluray.py", line 77, in __init__
    search_results = bluray_request('quicksearch/search.php', **params)['items']
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/flexget/plugins/internal/api_bluray.py", line 34, in bluray_request
    return requests.get(full_url, params=params).json(strict=False)
  File "/home/marco/flexget-dev/lib/python3.6/site-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.6/json/__init__.py", line 367, in loads
    return cls(**kw).decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2017-10-21 21:54 WARNING  task          movies-discover Aborting task (plugin: discover)
```

**Comment by cvium** [comment]
We don't support Python 3.6. Can you try Python 2.7 or 3.5?

**Comment by programatix** [comment]
I have to say that sometimes I get really frustrated with the python version requirement. I'm running FlexGet on my ASUS router using Entware. My option either Python 2.7.13 or 3.6.2. If I choose Python 2.7, I have this issue with the requirement on lib2to3 module. This module does not exists in the Python 2.7 package on mine due to the fact that lib2to3 module is actually meant for Python 3 to run Python 2 scripts. For this reason, I have requested before that scripts that import lib2to3 should detect the version of Python it is running on before importing it but so far nothing is done on it. Here's the log when I run Flexget on Python 2.
```
Traceback (most recent call last):
  File "/opt/bin/flexget", line 7, in <module>
    from flexget import main
  File "/opt/lib/python2.7/site-packages/flexget/__init__.py", line 12, in <module>
    from flexget.manager import Manager
  File "/opt/lib/python2.7/site-packages/flexget/manager.py", line 31, in <module>
    from flexget.utils.sqlalchemy_utils import ContextSession  # noqa
  File "/opt/lib/python2.7/site-packages/flexget/utils/sqlalchemy_utils.py", line 6, in <module>
    from past.builtins import basestring
  File "/opt/lib/python2.7/site-packages/past/__init__.py", line 88, in <module>
    from past.translation import install_hooks as autotranslate
  File "/opt/lib/python2.7/site-packages/past/translation/__init__.py", line 41, in <module>
    from lib2to3.pgen2.parse import ParseError
ImportError: No module named lib2to3.pgen2.parse
```

So, if I use Python 2.7, I always need to manually modify FlexGet to workaround this issue. If I use Python 3.6.2, whenever there is an exception, the first statement would be, it's not supported.

**Comment by girtsl** [comment]
This also happen with python 2.7:
```
2017-10-22 10:36 CRITICAL task          download-movies BUG: Unhandled error in plugin discover: Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flexget/task.py", line 486, in __run_plugin
    return method(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/event.py", line 23, in __call__
    return self.func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/input/discover.py", line 289, in on_task_input
    entries = self.estimated(entries, estimation_mode)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/input/discover.py", line 201, in estimated
    est_date = estimator.estimate(entry)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/estimators/est_released.py", line 28, in estimate
    estimate = estimator(entry)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/estimators/est_released_movies_bluray.py", line 36, in estimate
    movie = lookup(title=movie_name, year=movie_year, session=session)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/internal/api_bluray.py", line 209, in lookup
    updated_movie = BlurayMovie(title=title, year=year)
  File "<string>", line 4, in __init__
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/state.py", line 414, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/state.py", line 411, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/internal/api_bluray.py", line 77, in __init__
    search_results = bluray_request('quicksearch/search.php', **params)['items']
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/internal/api_bluray.py", line 34, in bluray_request
    return requests.get(full_url, params=params).json(strict=False)
  File "/usr/local/lib/python2.7/dist-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python2.7/dist-packages/simplejson/__init__.py", line 522, in loads
    return cls(encoding=encoding, **kw).decode(s)
  File "/usr/lib/python2.7/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/estimators/est_released.py", line 28, in estimate
    estimate = estimator(entry)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/estimators/est_released_movies_bluray.py", line 36, in estimate
    movie = lookup(title=movie_name, year=movie_year, session=session)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/internal/api_bluray.py", line 209, in lookup
    updated_movie = BlurayMovie(title=title, year=year)
  File "<string>", line 4, in __init__
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/state.py", line 414, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/state.py", line 411, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/internal/api_bluray.py", line 77, in __init__
    search_results = bluray_request('quicksearch/search.php', **params)['items']
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/internal/api_bluray.py", line 34, in bluray_request
    return requests.get(full_url, params=params).json(strict=False)
  File "/usr/local/lib/python2.7/dist-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python2.7/dist-packages/simplejson/__init__.py", line 522, in loads
    return cls(encoding=encoding, **kw).decode(s)
  File "/usr/lib/python2.7/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/lib/python2.7/dist-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
After running the same command a second time I get a different bug:
```
osmc@osmc:~$ flexget execute --discover-now --no-cache --tasks download-movies --debug --verbose
There is a FlexGet process already running for this config, sending execution there.
2017-10-22 10:39 INFO     performance                   Enabling plugin and SQLAlchemy performance debugging
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flexget/__init__.py", line 42, in main
    manager.start()
  File "/usr/local/lib/python2.7/dist-packages/flexget/manager.py", line 310, in start
    client.handle_cli(self.args)
  File "/usr/local/lib/python2.7/dist-packages/rpyc/core/netref.py", line 196, in __call__
    return syncreq(_self, consts.HANDLE_CALL, args, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/rpyc/core/netref.py", line 71, in syncreq
    return conn.sync_request(handler, oid, *args)
  File "/usr/local/lib/python2.7/dist-packages/rpyc/core/protocol.py", line 441, in sync_request
    raise obj
ValueError: before_plugin has already been registered as event listener under name task.execute.before_plugin

========= Remote Traceback (1) =========
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/rpyc/core/protocol.py", line 305, in _dispatch_request
    res = self._HANDLERS[handler](self, *args)
  File "/usr/local/lib/python2.7/dist-packages/rpyc/core/protocol.py", line 535, in _handle_call
    return self._local_objects[oid](*args, **dict(kwargs))
  File "/usr/local/lib/python2.7/dist-packages/flexget/ipc.py", line 85, in exposed_handle_cli
    self.manager.handle_cli(options)
  File "/usr/local/lib/python2.7/dist-packages/flexget/manager.py", line 349, in handle_cli
    self.execute_command(command_options)
  File "/usr/local/lib/python2.7/dist-packages/flexget/manager.py", line 370, in execute_command
    fire_event('manager.execute.started', self, options)
  File "/usr/local/lib/python2.7/dist-packages/flexget/event.py", line 106, in fire_event
    result = event(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/event.py", line 23, in __call__
    return self.func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/plugins/cli/performance.py", line 66, in startup
    add_event_handler('task.execute.before_plugin', before_plugin)
  File "/usr/local/lib/python2.7/dist-packages/flexget/event.py", line 76, in add_event_handler
    raise ValueError('%s has already been registered as event listener under name %s' % (func.__name__, name))
ValueError: before_plugin has already been registered as event listener under name task.execute.before_plugin
```

**Comment by cvium** [comment]
@programatix We have to make sure that the bug reports are not related to Python 3.6. We all have limited time to fix these bugs and we'd rather not waste time chasing down bugs that are not really FlexGet related.

**Comment by marcoacarvalho** [comment]
Thanks for the support.
As a note, I'm running Flexget on top of Python 3.6 for a good amount of time but only now I started to get this error

**Comment by programatix** [comment]
Well, at least we have @girtsl to confirm that it also occurred on Python 2.7.

**Comment by programatix** [comment]
@cvium I have proposed some changes at https://github.com/PythonCharmers/python-future on the module used by FlexGet to check the Python version before importing lib2to3. Hopefully they are accepted but I can see that the project is not that active. The following 2 files are involved,
1. /past/translation/__init__.py
2. /libfuturize/fixes/__init__.py

Another method is to let FlexGet do the checking (Python runtime version) before importing ```past.builtins```. I tried doing so for a few files locally and it seems to work fine. However there are a lot of files which imported it. I'm not sure whether not importing ```past.builtins``` would break FlexGet on Python2. If I can get confirmation that it's ok, I'm happy to edit all of them and request a pull for the changes. The following code is used when I tried locally,

```
import sys
if sys.hexversion >= 0x3000000:
    from past.builtins import basestring, ...
```
or
```
import sys
if sys.version_info[0] >= 3:
    from past.builtins import basestring, ...
```
